### DISCUSSION TITLE
TJA file support for osu!taiko (mapping)

### DISCUSSION THREAD

**Discussion by TheFearlessDeath500** [ppy/osu#23238] [Ideas] [answered]
To put it simply, the TJA file format is mostly used for Taiko no Tatsujin simulators such as TJAPlayer3 and OpenTaiko. It is the main kind of format entirely made in text form, quite similar to BMS/BME, except way simpler. All the notes are snapped as well in every line, which makes things very easy for charters/mappers who want to snap notes correctly.

Since you could already import BMS/BME files into osu!’s stable editor (unsure if lazer also supports importing from other file formats), I personally wouldn’t see the issue into adding this functionality in.

From what I’ve seen, some people have did some backwards engineering on the OSU file format and tried converting it to TJA (notably osu2tja, which is deprecated), but it feels difficult and time-consuming just to import a file in using programs/scripts.

If it helps, documentation for the format itself can be found on this page in particular; https://github.com/269Seahorse/Better-taiko-web/blob/master/TJA-format.mediawiki

I might join the osu!dev Discord in a few to discuss more about this later, but I feel like this could be a great addition to lazer. If this was already planned, I apologize.

**Comment by Rekunan** [comment]
I think it would help if you also gave some TJA files as examples.

**Reply by TheFearlessDeath500** [reply]
Oh damn I should have done that beforehand... Will add some examples if needed

**Comment by peppy** [comment, marked as answer]
I think this would best be done in an external application due to the rare usage case.

I'm not against support being added to the editor, but it won't be something we implement as a priority.

**Reply by TheFearlessDeath500** [reply]
Actually, I was a bit curious about this, but what do you mean by external application? Notepad or something similar or?

**Reply by peppy** [reply]
Something you can run which does the conversion outside of osu!

**Comment by TheFearlessDeath500** [comment]
UPDATE: Incredibly late response, but in this message I've added an example on how a TJA file looks like. TJA can be converted easily to TXT, as it's just that. Performs incredibly well with the UTF-8 with BOM format.
[Saitama 2000.txt](https://github.com/ppy/osu/files/15226699/Saitama.2000.txt)
